### PR TITLE
fix windows ucrt library filename

### PR DESCRIPTION
**Pull request by flavorjones**
Fix the DLL naming issue described at https://github.com/rake-compiler/rake-compiler-dock/issues/69 and https://github.com/grpc/grpc/pull/30081

This PR will be three commits, and I'm using CI to provide signal after the second and third:

- [x] expand CI coverage to include more versions of Ruby including 3.1 on windows (ucrt)
- [x] add tests to build static libraries (these will fail with ucrt)
- [x] fix the file names in the Dockerfile

I'll leave it in draft mode until I get to the final commit.

### CODE REVIEW

**Comment by flavorjones** [comment]
Boom, got the failure we wanted: https://github.com/rake-compiler/rake-compiler-dock/runs/7046504346?check_suite_focus=true

File: Dockerfile.mri.erb [hunk 1]
```diff
@@ -217,7 +217,7 @@ RUN find /usr/local/rake-compiler/ruby/*mingw*/ -name rbconfig.rb | while read f
 # Raise Windows-API to Vista (affects ruby < 2.6 only)
 RUN find /usr/local/rake-compiler/ruby -name rbconfig.rb | while read f ; do sed -i 's/0x0501/0x0600/' $f ; done
 # Don't link to static libruby
-RUN find /usr/local/rake-compiler/ruby -name *msvcrt-ruby*.dll.a | while read f ; do n=`echo $f | sed s/.dll//` ; mv $f $n ; done
+RUN find /usr/local/rake-compiler/ruby -name lib*-ruby*.dll.a | while read f ; do n=`echo $f | sed s/.dll//` ; mv $f $n ; done
```
Comment (larskanis) [inline Dockerfile.mri.erb:220]: Ahh yes, I missed this bit.

File: test/rcd_test/Rakefile [hunk 2]
```diff
@@ -41,11 +41,16 @@ namespace "gem" do
 
       desc "Build the native gem for #{plat}"
       task plat => 'prepare' do
+        config = if ENV['RCD_TEST_CONFIG']
```
Comment (larskanis) [inline test/rcd_test/Rakefile:44]: Is this condition necessary? An ending "--" to the rake task shouldn't disturb.

**Review by larskanis — APPROVED** [review]
Thank you so much for investigating this issue! And for extending the tests!

File: Dockerfile.mri.erb (hunk 1 above)
Comment (flavorjones) [reply Dockerfile.mri.erb:220]: :+1:

File: test/rcd_test/Rakefile (hunk 2 above)
Comment (flavorjones) [reply test/rcd_test/Rakefile:44]: Yup, you're right, I'll simplify.

**Comment by flavorjones** [comment]
@larskanis Once this goes green again (after responding to your comment above) and we merge it, would you mind cutting a new patch release for the grpc folks? (I can push the gem, but I don't think I have privileges to push the OCI images to dockerhub.)

**Comment by larskanis** [comment]
Yes, will do.

**Comment by flavorjones** [comment]
Thank you!

**Pull request merged**